### PR TITLE
Add optional subscribe param to some database APIs

### PR DESCRIPTION
**Pull request by abitmore**
PR for #1819.

These APIs are updated:
- added a `subscribe` parameter with type `optional<bool>`
  - get_objects `( vector ids, optional<bool> subscribe )`
  - get_assets `( vector symbols_or_ids, optional<bool> subscribe )`
  - get_accounts `( vector names_or_ids, optional<bool> subscribe )`
  - lookup_accounts `( string lower_bound, int limit, optional<bool> subscribe )`
  - get_htlc `( htlc_id id, optional<bool> subscribe )`
- changed type of `subscribe` parameter to `optional<bool>`
  - get_full_accounts `( vector names_or_ids, optional<bool> subscribe )`

```
* @param subscribe @a true to subscribe to the queried objects;
*                  @a false to not subscribe;
*                  @a omit or null to subscribe or not subscribe according to current
*                     auto-subscription setting (see @ref set_auto_subscription)
```

### CODE REVIEW

**Comment by abitmore** [comment]
Added test cases. This PR is now ready for review.

**Comment by abitmore** [comment]
@sschiessl-bcp and @startailcoon: you may have interest.

File: libraries/app/database_api.cpp [hunk 1]
```diff
@@ -87,12 +87,15 @@ class database_api_impl : public std::enable_shared_from_this<database_api_impl>
 
       // Accounts
       account_id_type get_account_id_from_string(const std::string& name_or_id)const;
-      vector<optional<account_object>> get_accounts(const vector<std::string>& account_names_or_ids)const;
-      std::map<string,full_account> get_full_accounts( const vector<string>& names_or_ids, bool subscribe );
+      vector<optional<account_object>> get_accounts( const vector<std::string>& account_names_or_ids,
+                                                     optional<bool> subscribe )const;
+      std::map<string,full_account> get_full_accounts( const vector<string>& names_or_ids, optional<bool> subscribe );
```
Comment (pmconrad) [inline libraries/app/database_api.cpp]: Please wrap.

File: libraries/app/include/graphene/app/database_api.hpp [hunk 2]
```diff
@@ -156,11 +156,14 @@ class database_api
...
        * @param ids IDs of the objects to retrieve
+       * @param subscribe @a true to subscribe to the queried objects; @a false to not subscribe;
+       *                  @a null to subscribe or not subscribe according to current auto-subscription setting
+       *                  (see @ref set_auto_subscription)
        * @return The objects retrieved, in the order they are mentioned in ids
        *
        * If any of the provided IDs does not map to an object, a null variant is returned in its position.
        */
-      fc::variants get_objects(const vector<object_id_type>& ids)const;
+      fc::variants get_objects( const vector<object_id_type>& ids, optional<bool> subscribe = optional<bool>() )const;
```
Comment (pmconrad) [inline libraries/app/include/graphene/app/database_api.hpp]: Please wrap.

File: libraries/app/database_api.cpp (hunk 1 above)
Comment (abitmore) [reply libraries/app/database_api.cpp]: I thought 118 characters was fine..
This looks good in Firefox, but I'd admit it's not very nice in Chrome.

**Pull request merged**